bluemira/builders/EUDEMO/first_wall/wall.py

```diff
@@ -0,0 +1,231 @@
+        )
+        self.adjust_variable(
+            "x2", ob_radius, lower_bound=ob_radius * 0.98, upper_bound=ob_radius + 2
+        )
+        self.adjust_variable(
+            "dz", self.variables["dz"].value, lower_bound=-3, upper_bound=3
+        )
+
+
+class WallBuilder(OptimisedShapeBuilder):
+    """
+    Builder class for the wall of the reactor, this does not include the
+    divertor.
+    """
+
+    COMPONENT_WALL_BOUNDARY = "wall_boundary"
+
+    _required_params: List[str] = [
+        "plasma_type",
+        "R_0",  # major radius
+        "kappa_95",  # 95th percentile plasma elongation
+        "r_fw_ib_in",  # inboard first wall inner radius
+        "r_fw_ob_in",  # inboard first wall outer radius
+        "A",  # aspect ratio
+    ]
+    _required_config: List[str] = []
+    _params: Configuration
+    _default_runmode: str = "mock"
+
+    def __init__(
+        self,
+        params: Dict[str, Any],
+        build_config: BuildConfig,
+        keep_out_zone: BluemiraWire = None,
+    ):
+        # boundary should be set by run/mock/read, it is used by the build methods
+        self.boundary: BluemiraWire
+        # _keep_out_zone should be set by reinitialize
+        self._keep_out_zone: Optional[BluemiraWire]
+
+        super().__init__(params, build_config, keep_out_zone=keep_out_zone)
+
+    def reinitialise(
+        self, params: Dict[str, Any], keep_out_zone: BluemiraWire = None
+    ) -> None:
+        """
+        Initialise the state of this builder ready for a new run.
+        """
+        super().reinitialise(params)
+        self._keep_out_zone = keep_out_zone
+
+    def read(self):
+        """Read the result of the design problem from file."""
+        return super().read()
+
+    def run(self):
+        """
+        Run the design problem for wall builder.
+        """
+        super().run(keep_out_zone=self._keep_out_zone)
+        self.boundary = self._shape.create_shape()
+
+    def mock(self):
+        """
+        Create a basic shape for the wall's boundary.
+        """
+        self.boundary = self._shape.create_shape()
+
+    def build(self, **kwargs) -> Component:
+        """
+        Build the components.
+        """
+        component = super().build(**kwargs)
+        component.add_child(self.build_xz())
+        return component
+
+    def build_xz(self) -> Component:
+        """
+        Build the components in the xz plane.
+        """
+        component = Component("xz")
+        component.add_child(
+            PhysicalComponent(
+                self.COMPONENT_WALL_BOUNDARY,
+                BluemiraWire(self.boundary, label=self.COMPONENT_WALL_BOUNDARY),
+            )
+        )
+        component.plot_options.wire_options["color"] = BLUE_PALETTE["DIV"]
+        bm_plot_tools.set_component_plane(component, "xz")
+        return component
+
+    # TODO(hsaunders1904): 'height' is not a parameter for all shapes.
+    # Does it make sense to derive it within this class?
```

Comment:
I think this links to my point above about supporting derived / calculated values dynamically in the config. It might also be that some problems need an offset if e.g. this polyspline height ends up within the keep out zone.

Reply:
spun out in #765